### ISSUE TITLE
The config resets itself on invalid configuration and doesn't create a backup.

### ISSUE THREAD

**Issue by mibby** [Efnilite/Walk-in-the-Park#37]
Ran into this error testing latest dev commit, compiled as of commit https://github.com/Efnilite/Walk-in-the-Park/commit/00843b9cd55fd90d8e306fa46e0a4e8fd2251d14.

Instead of the config just complaining about invalid configuration, it reset itself and wiped all settings (such as sql password) without backing itself up at start-up. Also breaks itself from ``/witp reload`` if you try making config changes that it doesn't like post-initialization.

This was caused by trying to remove the commands on the new scores config section. Tried setting ``scores: {}``, tried just removing the command line, tried removing the score values, also tried setting the command line to ``''``. Everything errors and resets it. :(
```
[19:00:32] [Server thread/ERROR]: Cannot load plugins/WITP/config.yml
org.bukkit.configuration.InvalidConfigurationException: while parsing a block mapping
 in 'string', line 273, column 3:
      enabled: true
      ^
expected <block end>, but found '<block mapping start>'
 in 'string', line 285, column 5:
        50:
        ^

	at org.bukkit.configuration.file.YamlConfiguration.loadFromString(YamlConfiguration.java:59) ~[1.16.5-patched.jar: "]
	at org.bukkit.configuration.file.FileConfiguration.load(FileConfiguration.java:160) ~[1.16.5-patched.jar: "]
	at org.bukkit.configuration.file.FileConfiguration.load(FileConfiguration.java:128) ~[1.16.5-patched.jar: "]
	at org.bukkit.configuration.file.YamlConfiguration.loadConfiguration(YamlConfiguration.java:188) ~[1.16.5-patched.jar: "]
	at dev.efnilite.witp.util.config.ConfigUpdater.update(ConfigUpdater.java:41) ~[?:?]
	at dev.efnilite.witp.util.config.Configuration.<init>(Configuration.java:51) ~[?:?]
	at dev.efnilite.witp.WITP.onEnable(WITP.java:85) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[1.16.5-patched.jar: "]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[1.16.5-patched.jar: "]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[1.16.5-patched.jar: "]
	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:500) ~[1.16.5-patched.jar: "]
	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:414) ~[1.16.5-patched.jar: "]
	at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:467) ~[1.16.5-patched.jar: "]
	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:260) ~[1.16.5-patched.jar: "]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:946) ~[1.16.5-patched.jar: "]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:176) ~[1.16.5-patched.jar: "]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: org.yaml.snakeyaml.parser.ParserException: while parsing a block mapping
 in 'string', line 273, column 3:
      enabled: true
      ^
expected <block end>, but found '<block mapping start>'
 in 'string', line 285, column 5:
        50:
        ^

	at org.yaml.snakeyaml.parser.ParserImpl$ParseBlockMappingKey.produce(ParserImpl.java:570) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:158) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.parser.ParserImpl.checkEvent(ParserImpl.java:148) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:235) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:162) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:257) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:248) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:236) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:162) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.composer.Composer.getNode(Composer.java:95) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.composer.Composer.getSingleNode(Composer.java:119) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:150) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:490) ~[1.16.5-patched.jar: "]
	at org.yaml.snakeyaml.Yaml.load(Yaml.java:416) ~[1.16.5-patched.jar: "]
	at org.bukkit.configuration.file.YamlConfiguration.loadFromString(YamlConfiguration.java:57) ~[1.16.5-patched.jar: "]
	... 16 more
```

```
[19:33:19] [Server thread/INFO]: mibby issued server command: /witp reload
[19:33:19] [Server thread/ERROR]: null
org.bukkit.command.CommandException: Cannot execute command 'witp' in plugin WITP v2.1a - plugin is disabled.
	at org.bukkit.command.PluginCommand.execute(PluginCommand.java:37) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:172) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.dispatchCommand(CraftServer.java:806) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.PlayerConnection.handleCommand(PlayerConnection.java:2146) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.PlayerConnection.c(PlayerConnection.java:1949) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.PlayerConnection.a(PlayerConnection.java:1902) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.PacketPlayInChat.a(PacketPlayInChat.java:47) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.PlayerConnectionUtils.lambda$ensureMainThread$1(PlayerConnectionUtils.java:43) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.TickTask.run(SourceFile:18) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.MinecraftServer.bb(MinecraftServer.java:1218) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.MinecraftServer.executeNext(MinecraftServer.java:1211) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.IAsyncTaskHandler.executeAll(IAsyncTaskHandler.java:95) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1351) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1016) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:176) ~[1.16.5-patched.jar:git-Yatopia-ver/1.16.5-"124113a"]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

**Comment by Efnilite** [comment]
This is _another_ problem with the config updater I use, thought I had fixed it, but guess not

**Comment by Efnilite** [comment]
Fixed in 52f05c748f4ad2c7cf4b930b3bf3a1204485ce82
